### ISSUE TITLE
PolynomialNormalization missing attribute

### ISSUE THREAD

**Issue by neilzxu** [neulab/xnmt#240]
c6fac72d109de8b7dffa2c6b15514ff4589565ef seems to have introduced a bug with PolynomialNormalization; can be reproduced by running the standard.yaml config file in examples/. 

Call:
`python3 xnmt/xnmt_run_experiments.py examples/standard.yaml --dynet-gpu`

Produces this output (truncated):

```
> Training
   Epoch 0.1002: train_loss/word=7.026294 (words=9426, words/sec=1150.65, time=0-00:00:08)
   Epoch 0.2000: train_loss/word=6.703589 (words=18864, words/sec=1155.63, time=0-00:00:16)
   Epoch 0.3003: train_loss/word=6.518545 (words=28110, words/sec=1188.03, time=0-00:00:24)
   Epoch 0.4005: train_loss/word=6.392840 (words=37588, words/sec=1138.11, time=0-00:00:32)
   Epoch 0.5003: train_loss/word=6.307682 (words=46648, words/sec=1159.72, time=0-00:00:40)
   Epoch 0.6004: train_loss/word=6.225853 (words=55790, words/sec=1144.99, time=0-00:00:48)
   Epoch 0.7000: train_loss/word=6.154869 (words=65143, words/sec=1154.20, time=0-00:00:56)
   Epoch 0.8003: train_loss/word=6.088884 (words=74466, words/sec=1143.59, time=0-00:01:04)
   Epoch 0.9000: train_loss/word=6.024372 (words=83640, words/sec=1188.11, time=0-00:01:12)
   Epoch 1.0000: train_loss/word=5.964533 (words=93086, words/sec=1154.51, time=0-00:01:20)
   Traceback (most recent call last):
     File "xnmt/xnmt_run_experiments.py", line 154, in <module>
          sys.exit(main())
     File "xnmt/xnmt_run_experiments.py", line 118, in main
          training_regimen.run_epochs(exp_args["run_for_epochs"])
     File "/home/ziyux/installs/miniconda3/envs/dynet/lib/python3.6/site-packages/xnmt-0.0.1-py3.6.egg/xnmt/train.py", line 190, in run_epochs
          self.one_epoch()
     File "/home/ziyux/installs/miniconda3/envs/dynet/lib/python3.6/site-packages/xnmt-0.0.1-py3.6.egg/xnmt/train.py", line 241, in one_epoch
          self.dev_evaluation()
     File "/home/ziyux/installs/miniconda3/envs/dynet/lib/python3.6/site-packages/xnmt-0.0.1-py3.6.egg/xnmt/train.py", line 260, in dev_evaluation
          xnmt.xnmt_decode.xnmt_decode(model_elements=(self.corpus_parser, self.model), **self.decode_args)
     File "/home/ziyux/installs/miniconda3/envs/dynet/lib/python3.6/site-packages/xnmt-0.0.1-py3.6.egg/xnmt/xnmt_decode.py", line 111, in xnmt_decode
          output = generator.generate_output(src, i, forced_trg_ids=ref_ids)
     File "/home/ziyux/installs/miniconda3/envs/dynet/lib/python3.6/site-packages/xnmt-0.0.1-py3.6.egg/xnmt/generator.py", line 6, in generate_output
          generation_output = self.generate(*args, **kwargs)
     File "/home/ziyux/installs/miniconda3/envs/dynet/lib/python3.6/site-packages/xnmt-0.0.1-py3.6.egg/xnmt/translator.py", line 127, in generate
          output_actions, score = self.search_strategy.generate_output(self.decoder, self.attender, self.trg_embedder, dec_state, src_length=len(sents), forced_trg_ids=forced_trg_ids)
     File "/home/ziyux/installs/miniconda3/envs/dynet/lib/python3.6/site-packages/xnmt-0.0.1-py3.6.egg/xnmt/search_strategy.py", line 97, in generate_output
          new_set.append(self.Hypothesis(self.len_norm.normalize_partial(hyp.score, score[cur_id], len(new_list)),
     File "/home/ziyux/installs/miniconda3/envs/dynet/lib/python3.6/site-packages/xnmt-0.0.1-py3.6.egg/xnmt/length_normalization.py", line 73, in normalize_partial
          return (score_so_far * pow(new_len-1, self.m) + score_to_add) / pow(new_len, self.m)
   AttributeError   :    'PolynomialNormalization' object has no attribute 'm'
```

**Comment by neubig** [comment]
Any ideas @msperber ?

**Comment by msperber** [comment]
Yes, to fix this we need to make the decoder Serializable as well, and parse it by invoking initialize_if_needed(). I'll send a PR when I find some time.

**Comment by neubig** [comment]
Thanks! For the time being, the work around is just to remove the
PolynomialNormalization option from the decode: section.

**Comment by neubig** [comment]
@msperber Any word on this? This is stopping experiments...

**Comment by neubig** [comment]
Fixed by #241
